addons/account/models/account_payment.py

```diff
@@ -773,12 +778,11 @@ def build_query(move_table_and_alias, outstanding_account_ids, payments):
         # Separate inbound and outbound payments, as their outstanding accounts differ and need to be checked separately
         payments_inbound = self.filtered(lambda p: p.payment_type == 'inbound')
         payments_outbound = self.filtered(lambda p: p.payment_type == 'outbound')
-        if self.journal_id:
-            inbound_outstanding_account_ids = tuple(self.journal_id._get_journal_inbound_outstanding_payment_accounts().ids)
-            outbound_outstanding_account_ids = tuple(self.journal_id._get_journal_outbound_outstanding_payment_accounts().ids)
-        else:
-            inbound_outstanding_account_ids = tuple(self.company_id.account_journal_payment_debit_account_id.ids)
-            outbound_outstanding_account_ids = tuple(self.company_id.account_journal_payment_credit_account_id.ids)
+        all_journals = self.env['account.journal'].search([])
+        if payments_inbound:
+            inbound_outstanding_account_ids = all_journals.inbound_payment_method_line_ids.payment_account_id.ids + self.company_id.account_journal_payment_debit_account_id.ids
```

Comment:
Added `self.company_id.account_journal_payment_debit_account_id.ids` because the default outstanding account was not included by fetching only `all_journals.inbound_payment_method_line_ids.payment_account_id.ids`. So if an user made no changes to the journal, no outstanding account would be fetched with `inbound_payment_method_line_ids.payment_account_id`.